server/sonar-db-dao/src/main/java/org/sonar/db/rule/RuleDefinitionDto.java

```diff
@@ -307,13 +307,13 @@ public RuleDefinitionDto setPluginKey(@Nullable String pluginKey) {
 
   @Override
   public boolean equals(Object obj) {
-    if (!(obj instanceof RuleDto)) {
+    if (!(obj instanceof RuleDefinitionDto)) {
```

Comment:
this deserve a dedicated commit

this is not a fixup, this is a bug in existing code. A UT would even be welcome

Reply:
Indeed I'll add one.